channel/channel.go

```diff
@@ -0,0 +1,214 @@
+package channel
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// Class containing states and metadata, and exposing convenience methods.
+type Channel struct {
+	Id      types.Destination
+	MyIndex uint
+
+	OnChainFunding types.Funds
+
+	state.FixedPart
+	// Support []uint64 // TODO: this property will be important, and allow the Channel to store the necessary data to close out the channel on chain
+	// It could be an array of turnNums, which can be used to slice into Channel.SignedStateForTurnNum
```

Comment:
In general you'll need to store more the "support" of one state -- that's generally *more* than one state -- to be safe.

 In those channels who are programmed to only accept unanimous consensus (a full set of sigs), can we get away with keeping one state. 
 
 I've left the implementation of Support out, for now, because it is not necessary to read from it during funding (the off chain protocols we are concentrating on for now). Of course we do need to write to it, and read from it for closing or challenging a channel.  We can make an issue to revisit it.